packages/flutter_tools/lib/src/base/binding.dart

```diff
@@ -0,0 +1,52 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:meta/meta.dart';
+
+import 'logger.dart';
+import 'platform.dart';
+import 'terminal.dart';
+
+/// The base binding which provides shared dependencies for the rest of the
+/// tool.
+mixin BindingBase {
+  /// Retreive the current [BindingBase] instance.
+  ///
+  /// This will be null if it was never initialized.
+  static BindingBase instance;
+
+  /// Whether the tool is being run in verbose mode.
+  bool get verbose;
+
+  /// The platform the tool is running under.
+  Platform get platform => const LocalPlatform();
+
+  @mustCallSuper
+  void initializeBinding() {
+    instance = this;
+  }
+
+  // Logging APIs
+
+  /// The output preferences for the arg results and logger.
+  OutputPreferences get outputPreferences => OutputPreferences();
```

Comment:
This creates a new instance every time the getter is invoked. Is that the intention?

Reply:
Accident :)

Reply:
previously it was `const` but i changed it for some reason...